mobzy-systems/src/main/kotlin/com/mineinabyss/mobzy/systems/listeners/MobListener.kt

```diff
@@ -175,13 +174,147 @@ object MobListener : Listener {
             isCancelled = true
     }
 
+    /** Ride entities with [Rideable] component on right click. */
+    @EventHandler
+    fun PlayerInteractEntityEvent.rideOnRightClick() {
+        val gearyEntity = rightClicked.toGearyOrNull() ?: return
+        val modelEntity = rightClicked.toModelEntity() ?: return
+
+        gearyEntity.with { rideable: Rideable ->
+            val mount = modelEntity.mountHandler
+            if (player.isSneaking) return
+
+            mount.setSteerable(true)
+            mount.setCanCarryPassenger(rideable.canTakePassenger)
+
+            if (!mount.hasDriver()) mount.driver = player
+            else mount.addPassenger("p_${mount.passengers.size + 1}", player)
+
+            if (rideable.canTakePassenger && mount.passengers.size < rideable.maxPassengerCount) {
+                mount.addPassenger("p_${mount.passengers.size + 1}", player) // Adds passenger to the next seat
+            }
+        }
+    }
+
+    /** Controlling of entities with [Rideable.requiresItemToSteer] */
+    @EventHandler
+    fun EntityMoveEvent.onMountControl() {
+        val gearyEntity = entity.toGearyOrNull() ?: return
+        val mount = entity.toModelEntity()?.mountHandler ?: return
+        val player = (mount.driver ?: return) as? Player ?: return
+
+        //TODO Make mob move on its own if not holding correct item
+        gearyEntity.with { rideable: Rideable ->
+            if (rideable.requiresItemToSteer && player.inventory.itemInMainHand != rideable.steerItem?.toItemStack())
+                isCancelled = true
+        }
+    }
+
+    /** Tame entities with [Tamable] component on right click */
+    @EventHandler
+    fun PlayerInteractEntityEvent.tameMob() {
+        val mob = (rightClicked as LivingEntity)
+        val maxHealth = mob.getAttribute(Attribute.GENERIC_MAX_HEALTH)?.value
+        val gearyEntity = rightClicked.toGearyOrNull() ?: return
+        val modelEntity = rightClicked.toModelEntity() ?: return
+        val itemInHand = player.inventory.itemInMainHand
+
+        gearyEntity.with { tamable: Tamable, rideable: Rideable ->
+            if (tamable.isTamable && !tamable.isTamed && tamable.tameItem?.toItemStack() == itemInHand) {
+                val random = Random(1).nextDouble()
+                tamable.isTamed = true
+                tamable.owner = player.uniqueId
+                rightClicked.toGeary().getOrSetPersisting { tamable.isTamed; tamable.owner == player.uniqueId }
+                player.spawnParticle(
+                    Particle.HEART,
+                    rightClicked.location.apply { y += 1.5 },
+                    10,
+                    random,
+                    random,
+                    random
+                )
+                return
+            }
+
+            //TODO Heal mob if fed tameItem
+            if (tamable.isTamed && (mob.health != maxHealth) && itemInHand == tamable.tameItem?.toItemStack()) {
```

Comment:
All branches here check isTamed, can add one branch to the when statement that returns if isnt ex.
```kt
when {
    !tamable.isTamed -> return
    (mob.health != maxHealth) && itemInHand == tamable.tameItem?.toItemStack() -> ...
    ...
}
```
You can actually use this trick several times in one statement, just remember when goes top to bottom until the first statement is matched.